binding.cc

```diff
@@ -404,8 +406,17 @@ struct Database {
     return db_->ReleaseSnapshot(snapshot);
   }
 
-  void ReleaseIterator (uint32_t id) {
-    iterators_.erase(id);
+  void AttachIterator (uint32_t id, Iterator* iterator) {
+    iterators_[id] = iterator;
+  }
+
+  void DetachIterator (uint32_t id) {
+    {
+      // Prevent parallel erase() calls
+      std::lock_guard<std::mutex> lock(iteratorsMutex_);
```

Comment:
Might be safe to remove now (it's possible that the problem was not with parallel iterators, but with parallel ending and gc on a single iterator).